atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -329,15 +326,16 @@ public void put(TableReference tableRef, Map<Cell, byte[]> values, long timestam
     }
 
     private void putIfNotUpdate(
-            TableReference tableRef,
-            DbWriteTable table,
+            DbReadTable readTable,
+            DbWriteTable writeTable,
             List<Entry<Cell, Value>> batch,
             KeyAlreadyExistsException ex) {
         Map<Cell, Long> timestampByCell = Maps.newHashMap();
         for (Entry<Cell, Value> entry : batch) {
             timestampByCell.put(entry.getKey(), entry.getValue().getTimestamp() + 1);
         }
-        Map<Cell, Value> results = get(tableRef, timestampByCell);
+
+        Map<Cell, Value> results = extractResults(readTable, readTable.getLatestCells(timestampByCell, true));
```

Comment:
Good catch to get rid of the double batching that might occur here.  That said are we comfortable with the increased size of the read that will occur here?  According to the defaults, the write batches are broken into either 1000 rows or 2 * 1024 * 1024 bytes, and prior the reads that had occurred here would have been batch further down to 256 rows (so 4 batched reads) whereas now they occur as one.  As an aside this is only true for posgres as Oracle doesn't use client side batching for reads.

Overall I'm not sure of the implications of this change, and am interested in what the benchmarks indicate.